pyvex/const.py

```diff
@@ -94,14 +104,22 @@ class U16(IRConst):
     c_constructor = pvc.IRConst_U16
 
     def __init__(self, value):
-        self.value = value
+        self._value = value
 
     def __str__(self):
         return "0x%04x" % self.value
 
     @staticmethod
     def _from_c(c_const):
-        return U16(c_const.Ico.U16)
+        global _U16_POOL
+
+        val = c_const.Ico.U16
+        if val < 1024:
+            return _U16_POOL[val]
+        return U16(val)
+
+
+_U16_POOL = [ U16(i) for i in range(1024) ]
```

Comment:
maybe make it the constants -1024 through 1024, for here and below?

Reply:
They are unsigned... why do we want to have negative numbers?

Reply:
Because vex will often encode subtracts as add with a large constant